src/module/actor/sheet/spellcasting-dialog.ts

```diff
@@ -1,112 +1,118 @@
-import { AbilityString } from "@actor/types";
-import { ABILITY_ABBREVIATIONS } from "@actor/values";
+import { ActorPF2e } from "@actor";
 import { SpellcastingEntryPF2e } from "@item";
-import { MagicTradition } from "@item/spell/types";
-import { PreparationType } from "@item/spellcasting-entry/data";
+import { SpellcastingEntrySystemData } from "@item/spellcasting-entry/data";
+import { pick } from "@util";
 
-interface SpellcastingDialogOptions {
-    entry?: SpellcastingEntryPF2e;
-    callback: (result: SpellcastingDialogResult, $html: JQuery) => void;
-}
-
-export interface SpellcastingDialogResult {
-    spellcastingType: PreparationType;
-    tradition: MagicTradition | "";
-    ability: AbilityString | "";
-    flexible: boolean;
+function createEmptySpellcastingEntry(actor: ActorPF2e): Embedded<SpellcastingEntryPF2e> {
+    return new SpellcastingEntryPF2e(
+        {
+            name: "Untitled",
```

Comment:
Is this ever seen by the user?

Reply:
Its not, but typescript kicks and screams if we omit it.

Reply:
could just be an empty name honestly.

Reply:
foundry schema check will kick and scream if we don't